app/channel.go

```diff
@@ -3181,7 +3190,13 @@ func (a *App) setChannelsMuted(channelIDs []string, userID string, muted bool) (
 		a.invalidateCacheForChannelMembersNotifyProps(member.ChannelId)
 
 		evt := model.NewWebSocketEvent(model.WebsocketEventChannelMemberUpdated, "", "", member.UserId, nil)
-		evt.Add("channelMember", member.ToJson())
+
+		memberJSON, jsonErr := json.Marshal(member)
+		if jsonErr != nil {
+			mlog.Warn("Failed to encode channel member to JSON", mlog.Err(jsonErr))
+		}
+
+		evt.Add("channelMember", memberJSON)
```

Comment:
Does it make sense to add to the event in case marshaling fails? It's possible that the field is empty right?

Same for the rest of the cases.

Reply:
Sure. Let's file a ticket to keep track of this.

Reply:
[MM-37909](https://mattermost.atlassian.net/browse/MM-37909)